### PR TITLE
Start compas.cad package

### PR DESCRIPTION
**Pull request by Licini**
This is to layout the basic structure of compas.cad package.

1. added object3d class with functionality of storing local/world transformation, hosting and traversing it children/parent nodes.
2. added scene class as container of object3ds to be visualised conveniently as a whole

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.

### Checklist

1. [ ] Add the change to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading: `Added`, `Changed`, `Removed`.
1. [x] Run all tests on your computer (i.e. `invoke test`).
1. [ ] If you add new functions/classes, check that:
   1. [x] Are available on a second-level import, e.g. `compas.datastructures.Mesh`.
   1. [ ] Add unit tests (especially important for algorithm implementations).

### CODE REVIEW

File: src/compas/cad/scene.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from compas.cad import Object3D
+
+
+__all__ = ["Scene"]
+
+
+class Scene(Object3D):
```
Comment (tomvanmele) [inline src/compas/cad/scene.py]: somehow this feels wrong to me. i would think that a scene and the objects in it are profoundly different. is this coming from ThreeJs?

**Comment by gonzalocasas** [comment]
I think `cad` is really not the correct name. Sorry to complain about naming again thou, but there's nothing in this package that relates to CAD. And actually, CAD is such a generic term that we should not use it to indicate that something is a sort of base for all CADs, because it is likely that we can reuse this for non-CAD tooling, like Unity/Unreal and other gaming engines.

Also, as a sort of convention remark, I believe any PR that proposes an entirely new package should include much more documentation at the module level. This one does not even link the new module to the API reference index, so no docs will be displayed at all.

**Comment by Licini** [comment]
> I think `cad` is really not the correct name. Sorry to complain about naming again thou, but there's nothing in this package that relates to CAD. And actually, CAD is such a generic term that we should not use it to indicate that something is a sort of base for all CADs, because it is likely that we can reuse this for non-CAD tooling, like Unity/Unreal and other gaming engines.
> 
> Also, as a sort of convention remark, I believe any PR that proposes an entirely new package should include much more documentation at the module level. This one does not even link the new module to the API reference index, so no docs will be displayed at all.

My bad for the doc, was intending to get some feedback from you guys first and finish docs at very end, maybe I should have put this as draft PR.

You have very fair points about the name `cad`, we are so used to migrate familiar terms and mis-apply them unconsciously everywhere....  
How about something like `canvas` or `canvas3d` ?   Or any other suggestions?

**Comment by gonzalocasas** [comment]
`canvas` sounds good to me, but my background with JS might be a bias here, so, we probably need to sit together and go through a few options. Could you please add a card about this for the next dev meeting?

**Comment by Licini** [comment]
> `canvas` sounds good to me, but my background with JS might be a bias here, so, we probably need to sit together and go through a few options. Could you please add a card about this for the next dev meeting?

sounds good, yep let's do that!

File: src/compas/cad/scene.py (hunk 1 above)
Comment (Licini) [reply src/compas/cad/scene.py]: Embarrassingly yes, https://threejs.org/docs/index.html#api/en/scenes/Scene
The main reason is to use the get_children() function from object3d to traverse everything held under the scene. However other than that, these is not much needed to inherent from object3d.  We could also have them as two entirely separate classes, which seems more right to you?

**Comment by tomvanmele** [comment]
i don't think it should be `compas.canvas`. it will not be clear what the package is for. furthermore, it should provide a basis for more than just the artists. it should also formalise picking, selecting, and finding objects, and an interface for, for example, updating attributes, and so on. so `compas.cad` might not be that bad after all...

**Comment by tomvanmele** [comment]
https://docs.blender.org/manual/en/latest/scene_layout/index.html

**Pull request closed without merging**